src/github/repository.ts

```diff
@@ -49,6 +51,10 @@ export const deleteRepositoryWebhookHandler = async (context: WebhookContext, ji
 		transformRepositoryId(context.payload.repository?.id, context.gitHubAppConfig?.gitHubBaseUrl)
 	);
 
+	// Update the total number of repos, we can assume atleast 1 repo(just deleted repo) exists for the default value
+	const totalNumberOfRepos = ((subscription.totalNumberOfRepos || 1) - 1);
+	await subscription.update({ totalNumberOfRepos });
+	await RepoSyncState.deleteRepoForSubscription(subscription, context.payload.repository?.id);
```

Comment:
I find `context.payload.repository?.id` could be dangerous. If it is `undefined` we are at the mercy of sequlize to do proper job to avoid delete all repos.

Maybe do a defensive check, and only proceed when it is not empty.